cimetrics/upload.py

```diff
@@ -23,13 +23,24 @@ def publish(self):
             self.env.mongo_connection
         except KeyError:
             print(
-                "Results were not uploaded since METRICS_MONGO_CONNECTION env is not set"
+                "Results were not uploaded since METRICS_MONGO_CONNECTION env is not set."
             )
             return
 
         client = pymongo.MongoClient(self.env.mongo_connection)
-        db = client[self.env.mongo_db]
-        coll = db[self.env.mongo_collection]
+
+        db = None
+        coll = None
+        try:
+            db = client[self.env.mongo_db]
+            coll = db[self.env.mongo_collection]
+        except:
```

Comment:
Never except:, except Exception at worst, but here we should really be more precise. We don’t want to hide a permission issue for example.